Packs/NutanixHypervisor/Integrations/NutanixHypervisor/NutanixHypervisor.py

```diff
@@ -0,0 +1,952 @@
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy, auth=auth)
+
+    def fetch_incidents(self, auto_resolved: Optional[bool], resolved: Optional[bool], acknowledged: Optional[bool],
+                        severity: Optional[str], alert_type_ids: Optional[str], impact_types: Optional[str]):
+        return self._http_request(
+            method='GET',
+            url_suffix='alerts',
+            params=assign_params(
+                resolved=resolved,
+                auto_resolved=auto_resolved,
+                acknowledged=acknowledged,
+                severity=severity,
+                alert_type_uuid=alert_type_ids,
+                impact_types=impact_types
+            )
+        )
+
+    def get_nutanix_hypervisor_hosts_list(self, filter_: Optional[str], limit: Optional[int], page: Optional[int]):
+        try:
+            return self._http_request(
+                method='GET',
+                url_suffix='hosts',
+                params=assign_params(
+                    filter_criteria=filter_,
+                    count=limit,
+                    page=page
+                )
+            )
+        except DemistoException as e:
+            if e.message and 'Invalid filter criteria specified.' in e.message:
+                raise DemistoException(
+                    'Filter criteria given to command nutanix-hypervisor-hosts-list is invalid, or is not written in '
+                    'the correct format. Check your format is correct by looking in the description of filter argument '
+                    'in the command')
+            raise e
+
+    def get_nutanix_hypervisor_vms_list(self, filter_: Optional[str], offset: Optional[int], limit: Optional[int]):
+        try:
+            return self._http_request(
+                method='GET',
+                url_suffix='vms',
+                params=assign_params(
+                    filter=filter_,
+                    offset=offset,
+                    length=limit
+                )
+            )
+        except DemistoException as e:
+            if e.message:
+                if 'Unrecognized field' in e.message:
+                    raise DemistoException('Filter criteria given to command nutanix-hypervisor-vms-list is invalid.')
+                if 'General error parsing FIQL expression' in e.message:
+                    raise DemistoException(
+                        'Filter criteria given to command nutanix-hypervisor-vms-list is not written in the current '
+                        'format. Check the correct format by looing in the description of filter argument in the '
+                        'command.')
+            raise e
+
+    def nutanix_hypervisor_vm_power_status_change(self, uuid: str, host_uuid: Optional[str], transition: str):
+        return self._http_request(
+            method='POST',
+            url_suffix=f'vms/{uuid}/set_power_state',
+            headers=self.CONTENT_JSON,
```

Comment:
Reason i didnt instantiate client with this header was that it will send this header in every request here, despite only 2 of the commands need this header. Tried to save network cost for unneeded sent header.
if this is no such big deal, ill instantiate client with this
@ShahafBenYakir

Reply:
ill add it to client instantiation